fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/v2/scanner/ScannerExtendedInfoReaderHelper.kt

```diff
@@ -28,79 +24,73 @@ import com.simprints.fingerprint.infra.scanner.v2.domain.root.responses.GetExten
 import com.simprints.fingerprint.infra.scanner.v2.domain.root.responses.GetHardwareVersionResponse
 import com.simprints.fingerprint.infra.scanner.v2.domain.root.responses.GetVersionResponse
 import com.simprints.fingerprint.infra.scanner.v2.domain.root.responses.SetVersionResponse
-import com.simprints.fingerprint.infra.scanner.v2.scanner.errorhandler.ResponseErrorHandler
-import com.simprints.fingerprint.infra.scanner.v2.scanner.errorhandler.handleErrorsWith
-import io.reactivex.Single
-import kotlinx.coroutines.rx2.await
-import kotlinx.coroutines.rx2.rxSingle
+
 
 class ScannerExtendedInfoReaderHelper(
     private val mainMessageChannel: MainMessageChannel,
     private val rootMessageChannel: RootMessageChannel,
-    private val responseErrorHandler: ResponseErrorHandler
 ) {
 
-    fun readScannerInfo(): Single<ScannerInformation> {
-        return getCypressVersion().flatMap { cypressVersion ->
-            val isLegacyApi = cypressVersion.apiMajorVersion <= CYPRESS_HIGHEST_LEGACY_API_MAJOR_VERSION
-                && cypressVersion.apiMinorVersion <= CYPRESS_HIGHEST_LEGACY_API_MINOR_VERSION
-
-            rxSingle { readScannerInfoBasedOnApiVersion(isLegacyApi) }
-        }
+    suspend fun readScannerInfo(): ScannerInformation {
+        val cypressVersion = getCypressVersion()
+        val isLegacyApi =
+            cypressVersion.apiMajorVersion <= CYPRESS_HIGHEST_LEGACY_API_MAJOR_VERSION && cypressVersion.apiMinorVersion <= CYPRESS_HIGHEST_LEGACY_API_MINOR_VERSION
+        return readScannerInfoBasedOnApiVersion(isLegacyApi)
     }
 
     private suspend fun readScannerInfoBasedOnApiVersion(isLegacyApi: Boolean): ScannerInformation {
         return if (isLegacyApi) getScannerInformationWithLegacyApi()
         else getScannerInformationWithNewApi()
     }
 
-    fun getCypressVersion(): Single<CypressFirmwareVersion> {
-        return sendRootModeCommandAndReceiveResponse<GetCypressVersionResponse>(
+    suspend fun getCypressVersion(): CypressFirmwareVersion {
```

Comment:
It could be a statement body to simplify it visually (the same applies to most other methods in the file).

```
suspend fun getCypressVersion(): CypressFirmwareVersion =  rootMessageChannel
   .sendRootModeCommandAndReceiveResponse<GetCypressVersionResponse>(GetCypressVersionCommand())
   .version
```